### PR TITLE
[MCKIN-10190] Report deletion errors

### PR DESCRIPTION
**Pull request by viadanna**
This PR modifies the API endpoints for user deletion to report failures.

**Testing instructions**:

1. Checkout this branch and restart the LMS making sure Python prereqs are reinstalled.

2. Stop the forum service to cause a failure during deletion.

3. Make a request to delete an existing user, e.g.:
```bash
$ curl -v -X DELETE -H "X-Edx-Api-Key: edx_api_key" localhost:8000/api/server/users?ids=<existing_user_id>
```

4. Make sure the response has the user email as key and the `HTTPConnectionPool` exception as error.

5. Check the LMS that the user still exists.

**Reviewers**
- [ ] @giovannicimolin 
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @viadanna! I've created [OSPR-3210](https://openedx.atlassian.net/browse/OSPR-3210) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by viadanna** [comment]
@giovannicimolin This is good for review. I fixed the tests but CircleCI is acting up with a 504 Gateway Time-out :(

**Comment by viadanna** [comment]
@Agrendalath This is the edxapp portion of the changes we discussed.

File: openedx/core/djangoapps/user_api/accounts/api.py [hunk 1]
```diff
@@ -559,21 +559,37 @@ def delete_users(users):
...
+    failed = {}
     for user in users:
-        retire_user_comments(user)
+        try:
+            retire_user_comments(user)
+        except Exception as e:
+            failed[user.email] = str(e)
 
     # Delete user profile images in background task
     usernames = list(users.values_list('username', flat=True))
```
[the remark this answers is not in the document]
Comment (viadanna) [reply openedx/core/djangoapps/user_api/accounts/api.py]: Given we have user data distributed along three different systems, there isn't much we can do about that. The idea here is keeping the `edxapp` user, i.e. not carrying deletion, so we can find out why it's failing, fixing it, and finally deleting it. This prevents orphaned data when errors happen.

**Comment by viadanna** [comment]
@giovannicimolin That shouldn't return a 500 in any case. I'll investigate as I couldn't replicate this in my devstack.

File: openedx/core/djangoapps/user_api/accounts/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply openedx/core/djangoapps/user_api/accounts/api.py]: Thanks for clarifying things for me. :grin:

**Pull request merged**